include/deal.II/base/tensor.h

```diff
@@ -2206,6 +2206,65 @@ operator * (const Tensor<rank,dim>     &t,
 }
 
 
+namespace Utilities
+{
+  namespace MPI
+  {
+    /**
+     * Perform an MPI sum of the entries of a tensor.
+     *
+     * @relates Tensor
+     */
+    template <int rank, int dim, typename Number>
+    inline
+    Tensor<rank,dim,Number>
+    sum (const Tensor<rank,dim,Number> &local,
+         const MPI_Comm &mpi_communicator)
+    {
+      unsigned int n_entries = local.n_independent_components;
+      std::vector<Number> entries( n_entries );
+
+      for (unsigned int i=0; i< n_entries; ++i)
+        entries[i] = local[ local.unrolled_to_component_indices(i) ];
+
+      std::vector<Number> global_entries( n_entries );
+      sum( entries, mpi_communicator, global_entries );
+
+      Tensor<rank,dim> global;
+      for (unsigned int i=0; i< n_entries; ++i)
+        global[ global.unrolled_to_component_indices(i) ] = global_entries[i];
+
+      return global;
+    }
+
+    /**
+     * Perform an MPI sum of the entries of a Tensor<1,dim>.
+     * This is a specialization of the general template.
+     *
+     * @relates Tensor
+     */
+    template <int rank, int dim, typename Number>
+    inline
+    Tensor<1,dim,Number>
+    sum (const Tensor<1,dim,Number> &local,
+         const MPI_Comm &mpi_communicator)
```

Comment:
I don't particularly need the specialization.  The only reason that I could think of was that for the rank 1 tensor I can statically allocate memory rather than making an std::vector.  Is that worth it?

It is no problem to remove it, of course.